sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -151,10 +168,41 @@ final class OffsetWindowFunctionFrame(
     }
     inputIndex += 1
   }
+}
 
-  override def currentLowerBound(): Int = throw new UnsupportedOperationException()
+/**
+ * The fixed offset window frame calculates frames containing
+ * NTH_VALUE statements.
+ * The fixed offset window frame return the same value for all rows in the window partition.
+ */
+class FixedOffsetWindowFunctionFrame(
+    target: InternalRow,
+    ordinal: Int,
+    expressions: Array[OffsetWindowFunction],
+    inputSchema: Seq[Attribute],
+    newMutableProjection: (Seq[Expression], Seq[Attribute]) => MutableProjection,
+    offset: Int)
+  extends OffsetWindowFunctionFrameBase(
+    target, ordinal, expressions, inputSchema, newMutableProjection, offset) {
 
-  override def currentUpperBound(): Int = throw new UnsupportedOperationException()
+  var maybeRow: Option[UnsafeRow] = None
+
+  override def prepare(rows: ExternalAppendOnlyUnsafeRowArray): Unit = {
+    super.prepare(rows)
+    if (inputIndex >= 0 && inputIndex < input.length) {
+      val r = WindowFunctionFrame.getNextOrNull(inputIterator)
+      maybeRow = Some(r)
+    }
+  }
+
+  override def write(index: Int, current: InternalRow): Unit = {
+    if (maybeRow.isDefined) {
+      projection(maybeRow.get)
```

Comment:
not related to this PR, but need some help to understand the code. Why in the `if`  branch we update the `target` row, but in the `else` branch we update the `current` row?